File: Leetcode/arrays/448_FindAllDisappearedNumber/Solution.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for(int i = 0; i<nums.length; ++i){
            int idx = Math.abs(nums[i])-1;
            nums[idx] = -Math.abs(nums[idx]);
        }
        List<Integer> res = new ArrayList<>();
        for(int i = 0; i<nums.length; ++i){
            if(nums[i]>0)
                res.add(i+1);
        }
        return res;
    }
}

File: Leetcode/arrays/448_FindAllDisappearedNumber/Solution.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i in range(len(nums)):
            k = abs(nums[i])-1
            nums[k] = -abs(nums[k])
        return [i+1 for i, num in enumerate(nums) if num>0]
